### PR TITLE
feat: upgrade nuxt and dependencies

### PR DESCRIPTION
**Pull request by csandru-plenty**
## Why:

Closes: [AB#114430](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/114430)

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/components/MegaMenu/MegaMenu.vue [hunk 1]
```diff
@@ -28,7 +28,7 @@
       <slot />
     </div>
 
-    <nav v-if="viewport.isGreaterOrEquals('md')" ref="floatingRef">
+    <nav v-if="viewport.isGreaterOrEquals('md') && categoryTree.length > 0" ref="floatingRef">
```
Comment (maxiroellplenty) [inline apps/web/components/MegaMenu/MegaMenu.vue]: @csandru-plenty looks like this is causing a layout shift

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/components/ReviewsAccordion/ReviewsAccordion.vue [hunk 2]
```diff
@@ -187,13 +183,17 @@ async function fetchReviews() {
...
 
-const deleteReview = () => {
-  fetchReviews();
-  emits('on-list-change');
-};
-const maxVisiblePages = computed(() => (viewport.isGreaterOrEquals('lg') ? 10 : 1));
+const deleteReview = () => fetchReviews();
 
-onMounted(() => fetchReviews());
+onNuxtReady(() => fetchReviews());
```
Comment (rbedeoan-plenty) [inline apps/web/components/ReviewsAccordion/ReviewsAccordion.vue]: nuxtReady is fetching Reviews on client, should't we fetch reviews on server for SEO lighthouse score ?

File: apps/web/__tests__/support/pageObjects/ProductListPageObject.ts [hunk 3]
```diff
@@ -77,10 +77,7 @@ export class ProductListPageObject extends PageObject {
   }
 
   goToProduct() {
-    cy.intercept('/plentysystems/getProduct').as('getProduct');
-    this.products.first().click();
-    cy.wait('@getProduct');
-
-    return this
+    this.products.first().wait(2000).click().wait(2000);
```
Comment (maxiroellplenty) [inline apps/web/__tests__/support/pageObjects/ProductListPageObject.ts:80]: all these wait's cant be the solution to fix the tests :/

File: apps/web/components/MegaMenu/MegaMenu.vue [hunk 4]
```diff
@@ -39,7 +39,9 @@
           }
         "
       >
-        <li v-for="(menuNode, index) in categoryTree" :key="index">
+        <li v-if="categoryTree.length === 0" class="h-10"></li>
```
Comment (maxiroellplenty) [inline apps/web/components/MegaMenu/MegaMenu.vue:42]: I dont get why the categoryTree is not there when the mega menu renders. The tree should load during ssr with the init call. We cant lazyload/render the top level here

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

File: apps/web/components/MegaMenu/MegaMenu.vue (hunk 4 above)
Comment (csandru-plenty) [reply apps/web/components/MegaMenu/MegaMenu.vue:42]: **I'll try to explain bellow, as we have 2 scenarios:**
1. server-side rendering (on first-time visit, hard-reload)
    - wait to initialize required data (for server & client): ```app.vue -> setInitialDataSSR -> categoryTree```
    - after required data has been initialized, proceed with the rest of the app logic (which actually depends on initialization: session, user, basket, categories...)
2. client-side rendering (not first-time visit, no hard-reload)
    - because data has already being initialized and the client is aware of it's state, composables use existing state for their logic (unless a fresh data fetch is actually required)

Because of the above we find rendering fake (progressive) content in apps (eg Facebook), that's something we eventually have to consider imo...

**Nuxt error-prone scenario (missing expected data, hydration):**
1. server-side rendering (on first-time visit, hard-reload), eg for **userObj**
   - server fetch required composables data, without an actual await for **userObj** response
   - client composables data also initialized, and these depend on expected **userObj** data
   - because of the broken order of operations (**userObj** state from the server might not be available by the time client state needs/expects **userObj** to exist) app will crash, introduce data/hydration errors
2. client-side rendering (not first-time visit, no hard-reload), eg for **userObj**
    - client composables are trying to read their known version of **userObj** data
    - meanwhile server data might have changed, and composables are not aware of the changes
    
In a nutshell:

Data needs to initialize at the same time for sever&client.
When making a data state change, composables have to do it for server&client.
- this is critical for 'parent' composables, that have other composables depending on their data status 
- that means we need to set the value on server in something global (like a plugin or middleware), or defer it entirely to client with ```onMounted``` or ```onNuxtReady```

File: apps/web/__tests__/support/pageObjects/ProductListPageObject.ts (hunk 3 above)
Comment (csandru-plenty) [reply apps/web/__tests__/support/pageObjects/ProductListPageObject.ts:80]: Because BE data response&initialization time varies, 2 out of 3 times tests are failing without a cool-off period, so the (non optimal) solution was to wait a bit before moving on.

Comment (maxiroellplenty) [reply apps/web/__tests__/support/pageObjects/ProductListPageObject.ts:80]: will go with it for now

**Review by maxiroellplenty — APPROVED** [review]
